src/GridFS/Bucket.php

```diff
@@ -728,4 +745,53 @@ private function registerStreamWrapper(): void
 
         StreamWrapper::register(self::STREAM_WRAPPER_PROTOCOL);
     }
+
+    /**
+     * Create a stream context from
+     *
+     * @see StreamWrapper::setDefaultContextResolver()
+     * @see stream_context_create()
+     *
+     * @param string $path The full url provided to fopen(). It contains the filename.
+     *                     gridfs://database_name/collection_name.files/file_name
+     *
+     * @return array{collectionWrapper: CollectionWrapper, file: object}|array{collectionWrapper: CollectionWrapper, filename: string, options: array}|null
+     */
+    private function resolveStreamContext(string $path, string $mode): ?array
+    {
+        // The file can be read only if it belongs to this bucket
+        $basePath = $this->createPathForFile((object) ['_id' => '']);
+        if (! str_starts_with($path, $basePath)) {
+            return null;
+        }
+
+        $filename = urldecode(substr($path, strlen($basePath)));
+
+        if (str_contains($mode, 'r')) {
+            $file = $this->collectionWrapper->findFileByFilenameAndRevision($filename, -1);
```

Comment:
In order to have the same path for read and write, we have to pass the filename in the path. So it's impossible to read a file from its `_id`, the file must have a filename and only the last revision is accessible.

Reply:
I'm not sure how this is supposed to work. `filename` is not even a unique field in GridFS -- not on its own nor in conjunction with a an upload date (see: [GridFS: Indexes](https://github.com/mongodb/specifications/blob/master/source/gridfs/gridfs-spec.rst#indexes)).

If we're going to use a URI to uniquely identify a GridFS file, the last component will need to be some encoded form of the `fs.files` document's `_id` field.